src/neo4j_genai/retrievers/external/pinecone/types.py

```diff
@@ -13,7 +13,8 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 from __future__ import annotations
-from typing import Any, Callable, Optional
+
+from typing import Any, Callable, Dict, List, Optional, Union
```

Comment:
With the annotations import, I thought we do not have to use `Dict` and `List` from `typing`, builtin types are ok (`dict` and `list`). This is not the case?

Reply:
I believe we need to use the `typing` imports if we want to support Python 3.8 (https://stackoverflow.com/questions/39458193/using-list-tuple-etc-from-typing-vs-directly-referring-type-as-list-tuple-etc)

Reply:
With `from __future__ import annotations` at the top of the file we shouldn't need to, does this cause an error for you?

Reply:
Have just ran the unit and E2E tests using 3.8 and the `list` and `dict` imports and seems there are no issues so I've updated them to use the builtin types